packages/sdk/src/HopBridge.ts

```diff
@@ -411,37 +460,156 @@ class HopBridge extends Base {
     return isHTokenTransfer ? l2BridgeAddress : ammWrapperAddress
   }
 
+  public async populateSendApprovalTx (
+    tokenAmount: TAmount,
+    sourceChain: TChain,
+    isHTokenTransfer: boolean = false
+  ):Promise<any> {
+    sourceChain = this.toChainModel(sourceChain)
+    const spender = await this.getSendApprovalAddress(sourceChain, isHTokenTransfer)
+    const isNativeToken = this.isNativeToken(sourceChain)
+    if (isNativeToken) {
+      return null
+    }
+    let token
+    if (sourceChain.isL1) {
+      token = this.getL1Token()
+    } else {
+      token = this.getCanonicalToken(sourceChain)
```

Comment:
If `isHTokenTransfer` is `true`, then this needs to approve the `hToken` on L2, not the canonical token.

Reply:
Nice catch, updated with https://github.com/hop-protocol/hop/pull/370/commits/2c7bd9ced4270df02d3b333ceef9dac72001601f